src/stage1/analyze.cpp

```diff
@@ -1001,6 +1002,7 @@ bool calling_convention_allows_zig_types(CallingConvention cc) {
         case CallingConventionUnspecified:
         case CallingConventionAsync:
         case CallingConventionInline:
+        case CallingConventionPtxKernel:
```

Comment:
Zig types have no ABI layout, and since PtxKernel is not a zig-internal calling convention this should return false at least when compiling for nvptx.

Reply:
I'm not sure if it's feasible, but I'd really like to be able to have GPU code with Zig objects as argument, like slices.

PTX doesn't enforce one calling convention: https://docs.nvidia.com/cuda/parallel-thread-execution/index.html#abstracting-abi
You can annotate the function parameters if they need to be read from registers.
The ptx also contains the size of each argument. 
So it may generate .ptx files that would be hard to launch from C, but I think this should be handled separately.
I'm fine if the .ptx is only possible to use with a specific version of Zig, as I'm likely to embed it inside the executable anyway.